### PR TITLE
refactor(typescript): migrate FloatingAction tests and stories to typescript

### PR DESCRIPTION
**Pull request by ivanhdez28**

### CODE REVIEW

**Review by David-Cohen35 — APPROVED** [review]
looks good to me

File: src/components/FloatingAction/index.test.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import FloatingAction from '.';
 
 describe('FloatingAction component', () => {
   it('renders component (set all props)', () => {
-    render(<FloatingAction Icon={() => null} label="delete the internet" />);
+    render(<FloatingAction Icon={() => <></>} label="delete the internet" />);
```
Comment (reywright) [inline src/components/FloatingAction/index.test.tsx:8]: Is there a reason we're passing a fragment here now?

Comment (ivanhdez28) [reply src/components/FloatingAction/index.test.tsx:8]: it's expecting an `Element`, the alternative is importing any Icon and pass it there, but seeing that it was passing a null, a fragment made sense.

Comment (reywright) [reply src/components/FloatingAction/index.test.tsx:8]: Hmm... my sort of gut feeling is that when we create a component, specifying that it returns a `JSX.Element` makes sense. However, specifying a prop that accepts a JSX.Element seems like a bad idea unless we explicitly know we want that.

The change I'd prefer I think (happy to chat about it) is to change this back to null, and in `src/components/Icon/index.tsx`, change line `12` to say 

```
type Icon = React.ComponentType<IconProps>
```

**Review by reywright — CHANGES_REQUESTED** [review]
Requesting some small changes on this one if that's alright.

File: src/components/FloatingAction/index.test.tsx (hunk 1 above)
Comment (ivanhdez28) [reply src/components/FloatingAction/index.test.tsx]: i agree with that, updated.

**Review by reywright — APPROVED** [review]
LGTM, thanks for addressing updates.

**Pull request merged**